lib/translations_checker/checker.rb

```diff
@@ -0,0 +1,93 @@
+require "translations_checker/diff"
+require "translations_checker/issues_presenter"
+require "translations_checker/terminal_colours"
+require "translations_checker/indentation"
+require "translations_checker/git_diff"
+require "translations_checker/change_checker"
+
+module TranslationsChecker
+  class Checker
+    include TerminalColours
+
+    using Indentation
+
+    def file_issues(en_diff)
+      translated_locales.flat_map { |locale| check_translation(locale, en_diff) }
+    end
+
+    def issues
+      @issues ||= diffs_for_locale("en").flat_map(&method(:file_issues))
+    end
+
+    # :reek:TooManyStatements
+    # rubocop:disable Metrics/AbcSize, Metrics/MethodLength
+    def run
+      if issues.empty?
+        puts green("No translations issues detected.")
+        return
+      end
+
+      puts yellow("Please correct the following translations issues before pushing:")
+      puts
+      puts IssuesPresenter.new(diffs_for_locale("en").flat_map(&method(:file_issues)))
+      puts
+      puts example
+      puts
+      puts yellow("Don't forget to keep the .yml keys hierarchy\n")
+      exit 1
+    end
+    # rubocop:enable Metrics/AbcSize, Metrics/MethodLength
+
+    def self.run
```

Comment:
What about using introduced `Service` concern here as well for consistency?

Reply:
Agreed.